### PR TITLE
Microsoft Graph Mail Single User - Support Update of Refresh token from Integration Parameters

### PR DESCRIPTION
**Pull request by ShacharKidor**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [XSUP-20490](https://jira-hq.paloaltonetworks.local/browse/XSUP-20490).

## Description
To solve the auth problem described in this issue - [XSUP-20490](https://jira-hq.paloaltonetworks.local/browse/XSUP-20490), I've added the following:
1. **MicrosoftGraphListener Integration** - The refresh token from the integration parameters (current instance config) is passed to the MicrosoftClient as a new argument called `refresh_token_param`. 

2. **MicrosoftAPIModule** - The following logic was added to the `_oproxy_authorize` function: If for some reason the auth with the Oproxy server has failed, and the `refresh_token_param` exists, we will make another authentication try using the `refresh_token_param` (the token from the current client config). If the second try succeeded we will continue with its response. If it also failed - we will raise a suitable error msg and write to the logs.

3. **MicrosoftGraphListener** - Added the `msgraph-mail-auth-reset` command that allows the user to reset the integration context. @yuvalbenshalom, @yucohen - If you think it is unnecessary please LMK and I'll revert this change. 

4. Same change described in (1) was added to the following integrations: **Azure Log Analytics**,  **Microsoft Management Activity API (O365 Azure Events)** (same bug reproduced for them as well). 

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by ShacharKidor** [comment]
@yuvalbenshalom, @yucohen - Please start the review. 

* The change in **MicrosoftGraphMail** is temporary (I made it only for triggering the TPBs of this integration to make sure that they pass successfully on instances of both self-deployed and Oproxy auth). 
* *  Will update the RN of all the packs that use the MicrosoftAPIModule when we will finish the review process (to prevent conflicts).

**Review by yuvalbenshalom — CHANGES_REQUESTED** [review]
looks good, see my comments

**Review by yucohen — APPROVED** [review]
Looks good from my side🙃

File: Packs/MicrosoftManagementActivity/Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.py [hunk 1]
```diff
@@ -560,7 +562,13 @@ def main():
             elif not enc_key and not (certificate_thumbprint and private_key):
                 raise DemistoException('Key or Certificate Thumbprint and Private Key must be provided.')
 
-        refresh_token = get_integration_context().get('current_refresh_token') or refresh_token
+        # The Client gets both the refresh token from the integration context (that was saved there in
+        # previous executions) and the 'refresh_token_param' which is the refresh token from the integration parameters
+        # (i.e current instance configuration). The 'refresh_token_param' will be used if for some reason the
+        # authentication with the oproxy server will fail (in that case we will perform a second attempt using the
+        # refresh_token_param).
```
[the remark this answers is not in the document]
Comment (ShacharKidor) [reply Packs/MicrosoftManagementActivity/Integrations/MicrosoftManagementActivity/MicrosoftManagementActivity.py:569]: Definitely better! will change that, thanks.

**Review by yuvalbenshalom — APPROVED** [review]

**Pull request merged**